bucket4j/src/main/java/com/linecorp/armeria/server/throttling/bucket4j/TokenBucket.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.throttling.bucket4j;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import com.google.common.base.MoreObjects;
+
+/**
+ * Stores configuration of the Token-Bucket algorithm, comprised of multiple limits.
+ */
+public final class TokenBucket {
+
+    /**
+     * Returns a newly created {@link TokenBucketBuilder}.
+     */
+    public static TokenBucketBuilder builder() {
+        return new TokenBucketBuilder();
+    }
+
+    /**
+     * Returns a newly created {@link TokenBucket}. Computes a set of {@link BandwidthLimit} out of
+     * a comma-separated {@code specification} string that conforms to the following format,
+     * as per <a href="https://tools.ietf.org/id/draft-polli-ratelimit-headers-00.html">RateLimit Header Scheme for HTTP</a>:
+     * <pre>{@code
+     * <bandwidth limit 1>[, <bandwidth limit 2>[, etc.]]
+     * }</pre>
+     * The order of elements inside {@code specification} is not defined.
+     * For example:
+     * <ul>
+     *   <li>{@code 100;window=60;burst=1000, 50000;window=3600}</li>
+     * </ul>
+     *
+     * @param specification the specification used to create a {@link BandwidthLimit}
+     * @see TokenBucketSpec#parseTokenBucket(String)
+     */
+    public static TokenBucket of(String specification) {
+        return TokenBucketSpec.parseTokenBucket(specification);
+    }
+
+    /**
+     * Returns a newly created {@link TokenBucket} with a single simple {@link BandwidthLimit}.
+     * Specifies easy limitation of {@code limit} tokens per {@code period} time window.
+     * @param limit the bucket size - defines the maximum count of tokens which can be held by the bucket
+     *              and defines the speed at which tokens are regenerated in the bucket
+     * @param period the time window, during which the tokens will be regenerated
+     * @return Newly created {@link TokenBucket}
+     * @see BandwidthLimit#of(long, Duration)
+     */
+    public static TokenBucket of(long limit, Duration period) {
+        return new TokenBucket(BandwidthLimit.of(limit, period));
+    }
+
+    @Nonnull
```

Comment:
We don't use `@Nonnull`.
All fields and arguments are non-null by default.
Could you remove this annotation from this PR, please?

Reply:
Fixed